samza-core/src/test/java/org/apache/samza/serializers/model/TestSamzaObjectMapper.java

```diff
@@ -165,7 +165,7 @@ public void testDeserializeUnknownTaskModelField() throws IOException {
   /**
    * Given a {@link ContainerModel} JSON with neither a processor-id nor a container-id, deserialization should fail.
    */
-  @Test(expected = SamzaException.class)
```

Comment:
Can you help us understand why the exception type changed?

Reply:
I do agree this seemed a bit icky, will try and work around it. Basically they are wrapping exceptions in Jackson as standard practice to add further context to the exception. The code that does this is nested here [MapDeserializer](https://github.com/FasterXML/jackson-databind/blob/jackson-databind-2.12.1/src/main/java/com/fasterxml/jackson/databind/deser/std/MapDeserializer.java#L621) which calls [ContainerDeserializerBase](https://github.com/FasterXML/jackson-databind/blob/jackson-databind-2.12.1/src/main/java/com/fasterxml/jackson/databind/deser/std/ContainerDeserializerBase.java#L173)


The originator of this exception is here in Samza and states it is for 0.13 and 0.14 support is this throw still required? [reference](https://github.com/apache/samza/blob/master/samza-core/src/main/java/org/apache/samza/serializers/model/SamzaObjectMapper.java#L109)

Will continue to look into it.

Reply:
@kw2542 a question on the API here, for general use case for SamzaException when should we use this?  For example if we get a deserialization error generally or serialization error would we always want SamzaException to wrap the Jackson exception. Dont think this would happen now as they are allowed to throw JsonProcessingException. Therefore this exception is thrown inside a deserialize call so would follow the API more suitably if it was a Json exception. Or should we add a catch up and wrap any Jackson exception in a Samza Exception.

Reply:
Good point, I don't think we have to stick with SamzaException and it won't be necessary to wrap JsonProcessingException in SamzaException again.

Reply:
If we did want to achieve this I raised an [issue](https://github.com/FasterXML/jackson-databind/issues/3068) in Jackson that stopped the [WRAP_EXCEPTION](https://fasterxml.github.io/jackson-databind/javadoc/2.12/com/fasterxml/jackson/databind/DeserializationFeature.html#WRAP_EXCEPTIONS) feature from working. With this disabled and tested against the SNAPSHOT it worked as originally. So we could wait for 2.12.2 to come out and it would behave the same.

Reply:
Thank you for help driving the fix! Let's wait for their fix if it is not too long away.